src/Shell/Process.php

```diff
@@ -543,6 +543,14 @@ public function kill($signal = SIGTERM)
             return false;
         }
 
+        foreach ($this->pipes as $index => $pipe) {
+            if ($index === static::STDIN) {
+                continue;
+            }
+
+            stream_set_blocking($pipe, static::NON_BLOCKING);
+        }
+
```

Comment:
Refactor the duplicated code into a method.

```php
protected function setStreamBlocking($blocking = static::NON_BLOCKING, $pipes = null)
{
    $pipes = $pipes ?: $this->pipes;
    // ...
}
```

Reply:
done